TestApps/Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift

```diff
@@ -29,127 +34,96 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
-    if editingStyle == .delete {
-      propertiesCount -= 1
-      tableView.deleteRows(at: [indexPath], with: .automatic)
-    } else if editingStyle == .insert {
-      propertiesCount += 1
-      tableView.insertRows(at: [indexPath], with: .automatic)
-    }
-  }
-  
-  func isInsertRow(at indexPath: IndexPath) -> Bool {
-    return indexPath.section == kPropertiesSection && indexPath.row == tableView(tableView, numberOfRowsInSection: indexPath.section) - 1
+    let propertySection = getPropertySection(at: indexPath)
+    propertySection?.tableView(tableView, commit: editingStyle, forRowAt: indexPath)
   }
-  
-  func isPropertiesRowSection(_ section: Int) -> Bool {
-    return section == kPropertiesSection
-  }
-  
+
   override func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCellEditingStyle {
-    if isInsertRow(at: indexPath) {
-      return .insert
-    } else {
-      return .delete
+    if let propertySection = getPropertySection(at: indexPath) {
+      return propertySection.tableView(tableView, editingStyleForRowAt: indexPath)
     }
+    return .delete
   }
-  
+
   override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
     tableView.deselectRow(at: indexPath, animated: true)
-    if isInsertRow(at: indexPath) {
+    let propertySection = getPropertySection(at: indexPath)
+    if propertySection != nil && propertySection!.isInsertRow(indexPath) {
       self.tableView(tableView, commit: .insert, forRowAt: indexPath)
     }
   }
-  
+
   override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-    if isPropertiesRowSection(section) {
-      return propertiesCount + 1
-    } else {
-      return super.tableView(tableView, numberOfRowsInSection: section)
+    if let propertySection = getPropertySection(at: IndexPath(row: 0, section: section)) {
+      return propertySection.tableView(tableView, numberOfRowsInSection: section)
     }
+    return super.tableView(tableView, numberOfRowsInSection: section)
   }
-  
+
   override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-    if isPropertiesRowSection(indexPath.section) {
+    if getPropertySection(at: indexPath) != nil {
       return super.tableView(tableView, heightForRowAt: IndexPath(row: 0, section: indexPath.section))
-    } else {
-      return super.tableView(tableView, heightForRowAt: indexPath)
     }
+    return super.tableView(tableView, heightForRowAt: indexPath)
   }
 
   override func tableView(_ tableView: UITableView, indentationLevelForRowAt indexPath: IndexPath) -> Int {
-    if isPropertiesRowSection(indexPath.section) {
-      return super.tableView(tableView, indentationLevelForRowAt: IndexPath(row: 0, section: indexPath.section))
-    } else {
-      return super.tableView(tableView, indentationLevelForRowAt: indexPath)
-    }
+    return 0
```

Comment:
What is this function doing now?  Is it still needed?

Reply:
It is still needed - there's a crash if it's removed. I can add a comment.